pkg/kubelet/volumes.go

```diff
@@ -104,6 +104,11 @@ func (vh *volumeHost) GetHostName() string {
 	return vh.kubelet.hostname
 }
 
+// Returns the storage config dir
```

Comment:
Please do not add new functions to `VolumeHost`, it's complicated enough already - this interface is (and will be) implemented in many places.

It seems that the config dir is relevant to glusterfs and maybe to ceph. To pass the config directory to appropriate plugins you can add a parameter to `glusterfs.ProbeVolumePlugins(configDir string)`. In `kube-controller-manager` init, you will then call:

```
	allPlugins = append(allPlugins, glusterfs.ProbeVolumePlugins(config.ConfigDir)...)
```

(and you need to add `ConfigDir` to `struct VolumeConfiguration`). This requires some plumbing here and there, but it's IMO much cleaner than adding new method to `VolumeHost`.